test/os.go

```diff
@@ -27,6 +28,46 @@ func TempDir(t testing.TB, base string) string {
 	return CanonPath(t, dir)
 }
 
+// IsNotExist calls os.Stat and asserts that the entry does not exist
+func IsNotExist(t testing.TB, dir, fname string) {
+	t.Helper()
+	_, err := os.Stat(filepath.Join(dir, fname))
+	if errors.Is(err, os.ErrNotExist) {
+		return
+	}
+	assert.NoError(t, err, "stat error")
+
+	t.Fatalf("file should not exist: %s", fname)
+}
+
+// IsDir calls os.Stat and asserts that the entry is a directory
+func IsDir(t testing.TB, dir, fname string) {
+	t.Helper()
+	isDirOrFile(t, dir, fname, true)
+}
+
+// IsFile calls os.Stat and asserts that the entry is a file
+func IsFile(t testing.TB, dir, fname string) {
+	t.Helper()
+	isDirOrFile(t, dir, fname, false)
+}
+
+func isDirOrFile(t testing.TB, dir, fname string, isDir bool) {
```

Comment:
I am wondering if its worthwhile to import `stretchr/testify` considering it already implements these functions (and other common assertions), then we wouldn't need to maintain these helpers.

For example:
* https://github.com/stretchr/testify/blob/331c520966e81450aef3b50956ea4a5db5e0ccdf/require/require.go#L68
* https://github.com/stretchr/testify/blob/331c520966e81450aef3b50956ea4a5db5e0ccdf/require/require.go#L572
* https://github.com/stretchr/testify/blob/331c520966e81450aef3b50956ea4a5db5e0ccdf/require/require.go#L1378

However I understand that it's out of scope of this PR and would need refactoring of other tests.

Reply:
Agreed, the less of this kind of code we have to write/maintain, the better.